workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/TextControllerAsMutableState.kt

```diff
@@ -0,0 +1,33 @@
+@file:OptIn(WorkflowUiExperimentalApi::class)
+
+package com.squareup.workflow1.ui.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.snapshotFlow
+import com.squareup.workflow1.ui.TextController
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+import kotlin.reflect.KProperty
+
+@Composable public fun TextController.asMutableState(): MutableState<String> {
+  return remember { mutableStateOf(textValue) }.also { state ->
+    // Restart the effect if a different controller is passed in.
+    LaunchedEffect(this) {
```

Comment:
Given how sticky `remember` is, will `LaunchedEffect` ever actually see a new `TextController` instance? 

I guess it will, but it won't matter because `state` will be the one that was created by the original `TextController`.

Very confusing, need to tweak this I think.

Reply:
Seems like that should be fine? But when I had the rendering swap between different `TextController` instances, the swap button didn't work…

OOOoooh. `mutableState` persists, the new `LaunchedEffect ` starts, and the new `TextController` is immediately clobbered by it. Something like that.

Will play more.

Reply:
Oh hell yeah, easy fix.